### ISSUE TITLE
false positive for `unnecessary-lambda` detection

### ISSUE THREAD

**Issue by SamyCookie** [PyCQA/pylint#2845]
### Steps to reproduce
1. check this code with Pylint
```python3
foo = lambda bar=None, **kwargs: print(bar, kwargs)
foo2 = lambda **kwargs: foo(bar=42, **kwargs)
```

### Current behavior
logs: `PyLint: Lambda may not be necessary`

### Expected behavior
no logs

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.5.3 (default, Sep 27 2018, 17:25:39) 
[GCC 6.3.0 20170516]
```

**Comment by SamyCookie** [comment]
Thanks @PCManticore to fixing this !
